src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -660,6 +660,47 @@ def update_period(self):
         except SubprocessError as error:
             raise DashboardException(error, http_status_code=500, component='rgw')
 
+    def edit_realm(self, realm_name: str, default: bool, new_realm_name: str):
+        rgw_realm_edit_cmd = []
+        cmd_edit_realm_options = []
+        if new_realm_name == realm_name:
+            if str_to_bool(default):
+                rgw_realm_edit_cmd = ['realm', 'default']
+                cmd_edit_realm_options = ['--rgw-realm', realm_name]
+                rgw_realm_edit_cmd += cmd_edit_realm_options
+                try:
+                    exit_code, _, err = mgr.send_rgwadmin_command(rgw_realm_edit_cmd)
+                    if exit_code > 0:
+                        raise DashboardException(e=err, msg='Unable to set {} as default realm'.format(realm_name),  # noqa E501  #pylint: disable=line-too-long
+                                                 http_status_code=500, component='rgw')
+                except SubprocessError as error:
+                    raise DashboardException(error, http_status_code=500, component='rgw')
+            else:
+                raise DashboardException(msg='The realm already exists',
+                                         http_status_code=400, component='rgw')
+        else:
+            rgw_realm_edit_cmd = ['realm', 'rename']
+            cmd_edit_realm_options = ['--rgw-realm', realm_name, '--realm-new-name', new_realm_name]
+            rgw_realm_edit_cmd += cmd_edit_realm_options
+            try:
+                exit_code, _, err = mgr.send_rgwadmin_command(rgw_realm_edit_cmd)
+                if exit_code > 0:
+                    raise DashboardException(e=err, msg='Unable to edit realm',
+                                             http_status_code=500, component='rgw')
+            except ValueError:
+                pass
+
+            if str_to_bool(default):
+                rgw_realm_edit_cmd = ['realm', 'default']
+                cmd_edit_realm_options = ['--rgw-realm', new_realm_name]
+                try:
+                    exit_code, _, _ = mgr.send_rgwadmin_command(rgw_realm_edit_cmd)
+                    if exit_code > 0:
+                        raise DashboardException(msg='Unable to set {} as default realm'.format(new_realm_name),  # noqa E501  #pylint: disable=line-too-long
+                                                 http_status_code=500, component='rgw')
+                except SubprocessError as error:
+                    raise DashboardException(error, http_status_code=500, component='rgw')
```

Comment:
Looking at this, shouldn't it be possible to update a realm's name and set the default at once?
If not, at the very least we should have some kind of indication for the user.

Right now, if I edit the name and set default, it will only edit the name.

Reply:
i had the same concern while testing yesterday.

Reply:
I've fixed this issue now. Thanks for noticing & bringing it up @Pegonzal @nizamial09

Reply:
I'm sorry @avanthakkar, but I don't see how this got fixed. 

If I edit the name and set as default, only the name will change.
[screen-capture (6).webm](https://user-images.githubusercontent.com/106810838/226552269-a2b23f0c-0111-43b8-afc7-f4eafdf9f0a9.webm)


Furthermore, if removing the default of a realm is not an option the checkbox shouldn't be editable as this causes a confusing behaviour for the user. We are letting the user edit the value but not save it.
[screen-capture (1).webm](https://user-images.githubusercontent.com/106810838/226551926-f05c32b6-e6f9-46c0-ba72-2e642600211f.webm)

Reply:
I am able to edit realm name and set as default too at same time.

Reply:
I tried it again, now it worked! Not sure why it didn't before,
anyways, thanks for taking care of that.